stingray/covariancespectrum.py

```diff
@@ -0,0 +1,428 @@
+            float as input, the same is used as the standard deviation which
+            is also used as the std. And if the std is an iterable of
+            numbers, their mean is used for the same purpose.
+
+
+        Attributes
+        ----------
+        energy_events : dictionary
+            A dictionary with energy bins as keys and time of arrivals of
+            photons with the same energy as value.
+
+        energy_covar : dictionary
+            A dictionary with mid point of band_interest and their covariance
+            computed with their individual reference band. The covaraince
+            values are normalized.
+
+        unnorm_covar : np.ndarray
+            An array of arrays with mid point band_interest and their
+            covariance. It is the array-form of the dictionary `energy_covar`.
+            The covariance values are unnormalized.
+
+        covar : np.ndarray
+            Normalized covaraiance spectrum.
+
+        covar_error : np.ndarray
+            Errors of the normalized covariance spectrum.
+
+        min_time : int
+            Time of arrival of the earliest photon.
+
+        max_time : int
+            Time of arrival of the last photon.
+
+        min_energy : float
+            Energy of the photon with the minimum energy.
+
+        max_energy : float
+            Energy of the photon with the maximum energy.
+
+        """
+        self.avg_covar = True  # Set parameter to distinguish between parent
+                               # class and derived class.
+
+        self._init_vars(event_list, dt, band_interest, ref_band_interest, std)
+
+        self.segment_size = segment_size
+
+        nbins = int((self.max_time - self.min_time + 1) / self.segment_size)
+
+        for n in range(nbins):
+            tstart = self.min_time + n*self.segment_size
+            tend = self.min_time + self.segment_size*(n+1) - 1
+            indices = np.intersect1d(np.where(self.event_list_T[0] >= tstart),
+                                     np.where(self.event_list_T[0] <= tend))
+
+            # Set minimum and maximum values for the specified indices value
+            self._init_special_vars(T_start=indices[0], T_end=indices[-1]+1)
+
+            energy_events = {}
+
+            self._construct_energy_events(energy_events, T_start=indices[0],
+                                          T_end=indices[-1]+1)
+
+            self._update_energy_events(energy_events)
+
+            energy_covar = {}
+
+            self._construct_energy_covar(energy_events, energy_covar)
+
+            if n == 0:
+                self.energy_covar = energy_covar
+            else:
+                for key in energy_covar.keys():
+                    self.energy_covar[key] = self.energy_covar.get(key, 0) + energy_covar[key]
+
+        # Now divide with total number of bins for averaging
+        for key, value in self.energy_covar.items():
+            self.energy_covar[key] /= nbins
+
```

Comment:
Do you construct the error for the averaged covariance somewhere? It might be useful to give the `_calculate_covariance_error` method in `Covariancespectrum` a keyword `M=1`, such that `AveragesCovariancespectrum` can call it with its own `M = number of segments`.